### PR TITLE
Vis network

### PR DESCRIPTION
**Pull request by nojaf**
First working attempt to make https://github.com/fsprojects/fantomas-tools/issues/538 work.
@jindraivanek could you review this to ensure all existing functionality remains?

### CODE REVIEW

**Comment by jindraivanek** [comment]
Code looks good to me. Will give it a try.

**Comment by jindraivanek** [comment]
Tried it, its working great, except scaling - all nodes seems to have same size.

**Comment by nojaf** [comment]
Alright, thanks for checking this out.
I'll look into the scaling problem. Mostly likely some wrong config.

File: src/client/fsharp/Oak/GraphView.fs [hunk 1]
```diff
@@ -166,43 +165,90 @@ let view =
...
+
+            let options: VisNetwork.options =
+                {| layout = layout
+                   interaction = {| hover = true |}
+                   width = $"{parentElement.clientWidth}"
+                   height = $"{parentElement.clientHeight}"
+                   nodes =
+                    {| scaling =
+                        {| label = scalingLabel
+                           min = minScaling |} |} |}
```
Comment (jindraivanek) [inline src/client/fsharp/Oak/GraphView.fs:239]: Removing `min` here solve scaling issue for me.

Comment (nojaf) [reply src/client/fsharp/Oak/GraphView.fs:239]: Hah, interesting. I found that I couldn't see anything without the min.
`minScaling` is used in two locations. Did you remove them both?

Comment (jindraivanek) [reply src/client/fsharp/Oak/GraphView.fs]: Just this one. Without second one its not working (tiny nodes).

Not sure what you mean, in old version `min` is not used.

But scaling still working weirdly, nodes are scaled up if you delve down to the tree (no collapsed node, but all nodes are big) or sometimes when change to free mode.

Probably scaling work differently here, and it need some experimenting with values.

Comment (nojaf) [reply src/client/fsharp/Oak/GraphView.fs]: Yes, I believe the settings are slightly different now.
`vis-react` was using `vis 4.x` while `vis-network` is `9.1.x` so we are probably trying to update a few majors.

**Comment by nojaf** [comment]
Thanks for the assist @jindraivanek 🙌.
I've changed `minScaling` to `10` as the font seemed to be a lot bigger locally than on the production site. We can always tweak this later.
I'm going to merge this one in, happy to revisit anything we may have missed!

**Pull request merged**